tests/envs/test_conveyorbelt_env.py

```diff
@@ -0,0 +1,81 @@
+"""Unit tests for the ConveyorBelt environment.
+
+This module tests the functionality of the ConveyorBeltEnv class, including:
+- Valid state transitions based on discrete actions (no-op, shift left, shift right)
+- Proper video recording during environment interaction
+- Correct insertion and shifting behavior of box values on the conveyor belt
+"""
+
+from pathlib import Path
+
+import numpy as np
+import pytest
+from gymnasium.wrappers import RecordVideo
+
+from hybrid_failure_discovery.envs.conveyorbelt_env import (
+    ConveyorBeltAction,
+    ConveyorBeltEnv,
+    ConveyorBeltState,
+)
+
+
+def test_conveyorbelt_env_with_state_checks():
+    """Detailed test of ConveyorBeltEnv with video recording and state
+    transition checks."""
+
+    env = ConveyorBeltEnv()
+    video_dir = Path("videos/test-conveyorbelt-env")
+    video_dir.mkdir(parents=True, exist_ok=True)
+
+    wrapped_env = RecordVideo(
+        env,
+        str(video_dir),
+        episode_trigger=lambda episode_id: episode_id == 0,
+    )
+
+    # Reset environment and get initial state
+    state, _ = wrapped_env.reset(seed=42)
+    assert isinstance(state, ConveyorBeltState)
+    current_values = state.values.copy()
+    print(f"\nInitial state values: {current_values}")
+
+    for _ in range(10):
+        # Sample random action directly (returns ConveyorBeltAction)
+        action = wrapped_env.action_space.sample()
```

Comment:
can you manually check whether this is reproducible by running the same test 3-5 times and printing the actions and seeing that it never changes? if it does change, we may need to do `wrapped_env.action_space.seed(seed=42)` at the top of this test function